uk.co.compucorp.civicrm.tasksassignments/js/test/services/task_test.js

```diff
@@ -0,0 +1,222 @@
+define([
+  'mocks/fabricators/task',
+  'mocks/data/task',
+  'tasks-assignments/app'
+], function (taskFabricator, taskMock) {
+  'use strict';
+
+  var taskMock = taskMock;
+  var fakeTask = taskFabricator.single();
+  var fakeTaskList = taskFabricator.list();
+  var fakeActivityTypes = taskFabricator.activityTypes();
+  var fakeTaskStatus = taskFabricator.taskStatus();
+  var fakeReminderNote = taskFabricator.reminderNote();
+
+  var mockBackendCalls = function ($httpBackend) {
+    $httpBackend.whenGET(/action=getoptions&entity=Document/).respond({});
+    $httpBackend.whenGET(/action=get&entity=CaseType/).respond({});
+    $httpBackend.whenGET(/action=get&entity=contact/).respond({});
+    $httpBackend.whenGET(/action=get&debug=true&entity=Task/).respond({});
+    $httpBackend.whenGET(/views.*/).respond({});
+  }
+
+  describe('Task', function () {
+    var Task, $httpBackend, $httpParamSerializer, requestBody, request = {};
+
+    beforeEach(module('civitasks.appDashboard'));
+
+    beforeEach(inject(function (_Task_, _$httpParamSerializer_, _$httpBackend_) {
+      Task = _Task_;
+      $httpBackend = _$httpBackend_;
+      $httpParamSerializer = _$httpParamSerializer_;
+
+      spyOn(Task, 'save').and.callThrough();
+    }));
+
+    beforeEach(function () {
+      $httpBackend.whenGET(/action=getoptions&debug=true&entity=Task/).respond({});
+      mockBackendCalls($httpBackend);
+    });
+
+    describe('save()', function () {
+      beforeEach(function () {
+        $httpBackend.whenPOST(/action=create&debug=true&entity=Task/).respond(function (method, url, data, headers) {
+          request.method = method;
```

Comment:
Why do you need to set all this,
request.method = method;
request.headers = headers;
request.data = data;
return [200];